cli/src/main.rs

```diff
@@ -0,0 +1,234 @@
+// #[macro_use]
+// extern crate clap;
+// use clap::App;
+
+use std::io::{stdin, stdout, StdoutLock, Write};
+use termion::input::TermRead;
+
+mod backend;
+mod conversation;
+mod identity;
+mod networking;
+mod user;
+
+const HELP: &'static str = "
+>>> Available commands:
+>>>     - update                                update the client state
+>>>     - reset                                 reset the server
+>>>     - register {client name}                register a new client
+>>>     - create group {group name}             create a new group
+>>>     - group {group name}                    group operations
+>>>         - send {message}                    send message to group
+>>          - invite {client name}              invite a user to the group
+>>          - update                            update the client state
+
+";
+
+fn update(client: &mut user::User, group_id: Option<String>, stdout: &mut StdoutLock) {
+    let messages = client.update(group_id).unwrap();
+    stdout.write_all(b" >>> Updated client :)\n").unwrap();
+    if !messages.is_empty() {
+        stdout.write_all(b"     New messages:\n\n").unwrap();
+    }
+    messages.iter().for_each(|m| {
+        stdout
+            .write_all(format!("         {}\n", m).as_bytes())
+            .unwrap();
+    });
+    stdout.write_all(b"\n").unwrap();
+}
+
+fn main() {
+    pretty_env_logger::init();
+
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
+    let stdin = stdin();
+    let mut stdin = stdin.lock();
+
+    stdout
+        .write_all(b" >>> Welcome to the MLS CLI :)\nType help to get a list of commands\n\n")
+        .unwrap();
+    let mut client = None;
+
+    loop {
```

Comment:
For future iterations: It would be nice to have an event loop here that polls new messages regularly.